app/routes.py

```diff
@@ -1,2 +1,237 @@
+
+            db.session.commit()
+            return make_response(video.to_json_video()), 200
+    
+    elif request.method == "DELETE":
+        if video == None:
+            return make_response("",404)        
+        else:
+            db.session.delete(video)
+            db.session.commit()
+            return make_response ({"id": video.video_id}), 200
+
+# This route will check out a video to a customer, and update the data in the database 
+@rental_bp.route("/check-out", methods=["POST"])
+def checkout_video():
+    rental = request.get_json()
+## Required Request Body Parameter
+
+#check customer ID and video ID if int
+    if not isinstance(rental["customer_id"], int) or not isinstance(rental["video_id"], int):
+        return {
+            "detail": "Invaid Data. Must be an Integer."
+        }, 400
+
+    rental_list = Rental(customer_id = rental["customer_id"],
+                        video_id = rental["video_id"],
+                        due_date = datetime.now() + timedelta(days=7))# return in  7 days to datetime.now(current_date)
+
+
+# 404: Not Found if the customer does not exist
+    customer = Customer.query.get(rental_list.customer_id)
+    if customer is None:
+        return "Not Found", 404
+
+# 404: Not Found if the video does not exist
+    video = Video.query.get(rental_list.video_id)
+    if video is None:
+        return "Not Found", 404
+
+    if video.available_inventory <= 0:
+        return {
+            "message": "Invalid data."
+        }, 400
+
+    customer.checkout_count += 1
+    video.available_inventory -= 1
+
+
+    db.session.add(rental_list)
+    db.session.commit()
+
+    return jsonify({
+        "customer_id": rental_list.customer_id,
+        "video_id": rental_list.video_id,
+        "due_date": rental_list.due_date, #7 days from checked out date
+        "videos_checked_out_count": customer.checkout_count,
+        "available_inventory": video.available_inventory 
+    }), 200
+
+
+
+
+# This route checks in a video to a customer, and updates the data in the database
+@rental_bp.route("/check-in", methods=["POST"])
+def customer_check_in():
+    request_body = request.get_json()
+
+    customer_id = request_body.get("customer_id")
+    video_id = request_body.get("video_id")
+    # if the request body is missing any piece 
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        return jsonify({
+                "detail": "Invaid Data."
+                }), 404
+    # verify the customer and video ids are both integers 
+    if not isinstance(request_body["customer_id"], int) or not isinstance(request_body["video_id"], int):
+        return {
+            "detail": "Invaid Data. Must be an Integer."
+        }, 404
+    
+    # accessing each video id -- rental and customer id 
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    for rental in customer.video:
```

Comment:
To access all the rentals for a particular customer, use your rentals relationship.```for rental in customer.rentals``` -- and the rest of your logic should work out.